### ISSUE TITLE
Files not linted if target name contains a space

### ISSUE THREAD

**Issue by erikkerber** [realm/SwiftLint#3076]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [ ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

If a target name contains a space, i.e. "Experiences - Account", SwiftLint will not lint any of the files associated with that target.

<img width="157" alt="image" src="https://user-images.githubusercontent.com/75236/73905312-84ca4780-4864-11ea-82de-f2be1131ba7c.png">

We like to "categorize" our targets like this, so if there's a space bug we'll find it!

The root cause is a mix of needed escaping/unescape handling in **CompilerArgumentsExtractor.swift**:  `allCompilerInvocations` (and potentially `expandingResponseFiles`)

##### Complete output when running SwiftLint, including the stack trace and command used

(Abridged since problem is known)

Expected ~4000 files total, not 1356:

```bash
$ swiftlint analyze --reporter emoji --compiler-log-path build_logs/xcode_build_raw.log *

Collecting 'Foo.swift' (1/1356)
Collecting 'Bar.swift' (2/1356)
Collecting 'Fizz.swift' (3/1356)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?

0.38.2

* Installation method used (Homebrew, CocoaPods, building from source, etc)?

Source

* Paste your configuration file:

```iml
opt_in_rules:
    - unused_declaration
    - unused_import
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  No
* Which Xcode version are you using (check `xcodebuild -version`)?

11.3

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

N/A
